### ISSUE TITLE
Value of type 'MessagingInApp' has no member 'dismissMessage'

### ISSUE THREAD

**Issue by beqramo** [customerio/customerio-reactnative#141]
<img width="1694" alt="image" src="https://github.com/customerio/customerio-reactnative/assets/40486471/a635841a-7322-4dd0-9ce7-701aa87d5031">
Guys please, I can't even build the app now.

version: 2.4.0

**Comment by mrehan27** [comment]
Hi @beqramo. Sorry for the inconvenience. `2.4.0` release requires a method from iOS SDK that was added in `2.6.0`. Running the following command in project root directory should fix this for you.

```
pod update --repo-update --project-directory=ios
```

Please let us know if the issue still exists or if you have more questions.

**Comment by mrehan27** [comment]
Hi @OtamNitram. Sorry to hear that. Can you please confirm if your `ios/Podfile.lock` was updated after running `pod update`? It should look something like follows:

```
  - customerio-reactnative (2.4.0):
    - CustomerIO/MessagingInApp (~> 2)
    - CustomerIO/Tracking (~> 2)
    - React-Core
- CustomerIO/MessagingInApp (2.6.1):
    - CustomerIOMessagingInApp (= 2.6.1)
  - CustomerIO/MessagingPushAPN (2.6.1):
    - CustomerIOMessagingPushAPN (= 2.6.1)
  - CustomerIO/Tracking (2.6.1):
    - CustomerIOTracking (= 2.6.1)
  - CustomerIOCommon (2.6.1)
  - CustomerIOMessagingInApp (2.6.1):
    - CustomerIOTracking (= 2.6.1)
    - Gist (= 3.2.2)
  - CustomerIOMessagingPush (2.6.1):
    - CustomerIOTracking (= 2.6.1)
  - CustomerIOMessagingPushAPN (2.6.1):
    - CustomerIOMessagingPush (= 2.6.1)
  - CustomerIOTracking (2.6.1):
    - CustomerIOCommon (= 2.6.1)
```

If your lock file doesn't look like above, please make sure to update `ios/Podfile` as suggested in [our docs](https://www.customer.io/docs/sdk/react-native/push/#integrate-push-notifications-in-your-app)

```ruby
 pod 'CustomerIO/MessagingPushAPN', '~> 2'
```

After updating, please re-run the the command in your project root and let us know if this doesn't fix the problem for you.
```
pod update --repo-update --project-directory=ios
```

**Comment by beqramo** [comment]
Hi,
I had this:
  pod 'CustomerIO/MessagingPushAPN', '~> 2.1.2'  
  
If you are changing such an important thing please add it to a changelog.
It isn't a correct thing to check your docs for installation every time you update the package.

I will change it and will post it here, Please keep in mind the above statement.
Thanks

**Comment by beqramo** [comment]
I will select 2.6.1 version as I don't want automatic updates to affect the app.

**Comment by beqramo** [comment]
<img width="547" alt="image" src="https://github.com/customerio/customerio-reactnative/assets/40486471/888317a3-a5e3-480d-ac3d-6dd73b17d133">
I can't install 2.6.1, only 2.6.0 is available for me

**Comment by mrehan27** [comment]
Thank you for sharing your concerns, @beqramo. We understand the importance of clear documentation and transparent updates, and I apologize if there was any confusion in this case.

We're definitely mindful of this, and I want to ensure we are discussing this internally to improve the experience. We appreciate your understanding and patience as we work to address your concerns.

**Comment by vikrant-k-simformsolutions** [comment]
Hi, @mrehan27 
I'm using this  "customerio-reactnative": "^2.2.0" in my app. I tried all the above solutions but not getting success. So is there any other option to solve this issue with the same version or do I need to update my customerio-reactnative?

**Comment by mrehan27** [comment]
Hi @vikrant-k-simformsolutions. We sincerely apologize for the inconvenience caused due to this. We have released a fix in `2.4.2` that should resolve the problem for you. Please update your SDK to the latest version and run `pod update` as suggested above, and you should no longer encounter this issue.

Thank you for bringing this to our attention, and we appreciate your patience.
